python/ray/serve/common.py

```diff
@@ -14,6 +15,17 @@
 Duration = float
 
 
+@dataclass
+class EndpointInfo:
+    http_methods: List[str]
+    python_methods: Optional[List[str]] = None
+    route: Optional[str] = None
+
+    def __post_init__(self):
+        if self.python_methods is None:
+            self.python_methods = []
+
```

Comment:
Use fields instead: https://docs.python.org/3/library/dataclasses.html#dataclasses.field

Reply:
Nice, I knew there must be a better way to do this!